packages/ui/src/utils/get-potential-path.test.ts

```diff
@@ -0,0 +1,65 @@
+import { getPotentialPath } from './get-potential-path';
+
+describe('getPotentialPath', () => {
+  describe('forward direction', () => {
+    it('should increment last element when it is integer and penultimate is not', () => {
+      expect(getPotentialPath('route.from.steps.0', 'forward')).toBe('route.from.steps.1');
+      expect(getPotentialPath('route.from.steps.5', 'forward')).toBe('route.from.steps.6');
+    });
+
+    it('should increment penultimate element when last is not integer and penultimate is', () => {
+      expect(getPotentialPath('route.from.steps.0.when', 'forward')).toBe('route.from.steps.1');
+      expect(getPotentialPath('route.from.steps.3.doCatch', 'forward')).toBe('route.from.steps.4');
+    });
+
+    it('should handle single element paths', () => {
+      expect(getPotentialPath('0', 'forward')).toBeUndefined();
+      expect(getPotentialPath('route', 'forward')).toBeUndefined();
+    });
+  });
+
+  describe('backward direction', () => {
+    it('should decrement last element when it is integer and penultimate is not', () => {
+      expect(getPotentialPath('route.from.steps.1', 'backward')).toBe('route.from.steps.0');
```

Comment:
How about `route.from.steps.0` & `backward`?

Reply:
We already have a set of scenarios that return `undefined`, grouped under the test case: 'should return undefined when attempting to go below 0'.